### ISSUE TITLE
Implement the fallback mechanism for themes

### ISSUE THREAD

**Issue by pkarw** [DivanteLtd/vue-storefront#1711]
* **What is the motivation/use case for changing the behavior or adding a new feature?**

This is a requirement from our partner Vendic. I believe it's a requirement about theme fallback as it's available in Magento:
- we have a custom theme
- we have a default theme
- components' not present in the custom are loaded from the default

So this is kind of inheritance intended to simplify the themes development and updateability.

* **Acceptance criteria**
- [ ] There will be a basic theme inherited by the default in the custom

**Comment by filrak** [comment]
I can take care of it but maybe we should do it during implementation of new default theme? RIght now it'll require to include all packaes that current default theme needs along with this fallback. It'll be much easier if we'll have this mechanism in mind while creating this new theme.

**Comment by pkarw** [comment]
New theme z good chance to do it

**Comment by dlandmann** [comment]
@filrak and @pkarw hi - is there any news or will this really be implemented with the new theme? This is currently a small blocker in our implementation :-(

**Comment by filrak** [comment]
@dlandmann it will be implemented in new theme but you can easily have such capabilities just by using webpack and importing files from old theme into new one.

Please let me know in which way it's blocking you (what exactly you need to achieve) and i can propose a working solution for you.

**Comment by pkarw** [comment]
Hey! As Filip said - I'd recommend you to start with your `themes/{theme}/router.ts` and just import the things you haven't overridden in the theme - directly from `default`. Take a look at how `default` theme is re-using some parts of `default-amp` (only the Product and Category pages for some specific URLs).

Another option is to use webpack aliases to virtually switch the components being used - kind of fallback can be implemented using this method

**Comment by dlandmann** [comment]
Very good! Thx for the suggestion! 

The idea with the Webpack Aliases is interesting. I'll try that. Is there an example somewhere?

We have (in Magento) a main theme and derived from it a subtheme which has several "subsubthemes" ...

**Comment by dlandmann** [comment]
@filrak and @pkarw -  thanks for the ideas. we have found a very smart and nice and super simple Solution.

FYI - 
We solved the "problem" that way. The Key is **npm-config**!

We start the build or dev process by setting two env variables. Client (mandant1, mandant2 etc.) and Environment. (development, production), Due to the inheritance of the config-files (npm-config), only the individual values ​​have to be set for the clients and can then be easily stored in the pages / components, e.g. be queried via "v-if =" etc.

Overview Configfiles in / config, sorted by load order:

1. default.json // core file
2. local.json // our global settings
3. local-mandant.json // Change according to client
4. local-environment-mandant.json // Changes according to environment

This works very well in our case, as we do not adjust very many values ​​from client to client.

We can use it to easily implement all our client pages with **one theme**!
